tests/subsys/debug/gdbstub/testcase.yaml

```diff
@@ -17,12 +18,40 @@ tests:
         - "--gdb_script"
         - "test_breakpoints.gdbinit"
         - "--gdb_target_remote"
-        - "tcp:127.0.0.1:5678"
+        - "tcp::5678"
     tags:
       - debug
       - gdbstub
       - pytest
     extra_configs:
       # Make sure the gdbstub port chosen is unique for this test to avoid conflicts
       # when Twister runs tests in parallel on the same host.
-      - CONFIG_QEMU_EXTRA_FLAGS="-serial tcp:127.0.0.1:5678,server"
+      - CONFIG_GDBSTUB=y
+      - CONFIG_GDBSTUB_SERIAL_BACKEND=y
+      - CONFIG_QEMU_EXTRA_FLAGS="-serial tcp::5678,server"
+
+  # Connect to QEMU gdbstub and run the same simple GDB script
+  # Use non-default QEMU gdbstub port 1235
+  debug.gdbstub_qemu.breakpoints:
```

Comment:
how does this scenario test the gdbn stub feature in zephyr? It seems to be testing general gdb connection here.

Reply:
This test suite runs the same GDB script on two gdbstub implementations: our main target Zephyr gdbstub, and Qemu gdbstub reference target.
The latter is not only a part of Zephyr debugging toolset, but here serves as a reference implementation of RDP backend, so we check the expected behavior is the same.

Reply:
ok